src/App.tsx

```diff
@@ -16,47 +16,87 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  enum SortType {
+    byLength,
+    byAlphabet,
+    empty,
+  }
+
+  const [sortValue, setSortValue] = useState<SortType>(SortType.empty);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const sortGoods = (good1: string, good2: string): number => {
+    switch (sortValue) {
+      case SortType.byLength:
+        return good1.length - good2.length;
+
+      case SortType.byAlphabet:
+        return good1.localeCompare(good2);
+
+      default:
+        return 0;
+    }
+  };
+
+  const getPreparedGoods = (goods: string[]): string[] => (isReversed
+    ? [...goods].sort(sortGoods).reverse()
+    : [...goods].sort(sortGoods));
+
+  const checkState = () => (sortValue !== SortType.empty
+    || isReversed !== false);
```

Comment:
It can be used as a variable. And use a more meaningful name. Here is my suggestion
```suggestion
  const isSortApplied = sortValue !== SortType.empty || isReversed;
```